packages/ra-ui-materialui/src/list/List.js

```diff
@@ -113,48 +111,27 @@ export const ListView = ({
                     className={classes.header}
                     title={titleElement}
                     actions={React.cloneElement(actions, {
-                        className: classes.actions,
-                    })}
-                    actionProps={{
                         ...controllerProps,
+                        className: classes.actions,
                         bulkActions,
                         exporter,
                         filters,
-                    }}
+                    })}
                 />
                 {filters &&
                     React.cloneElement(filters, {
                         ...controllerProps,
                         context: 'form',
                     })}
-                {isLoading || total > 0 ? (
-                    <div key={version}>
-                        {children &&
-                            React.cloneElement(children, {
-                                ...controllerProps,
-                                hasBulkActions: !!bulkActions,
-                            })}
-                        {!isLoading &&
-                            !rest.ids.length && (
-                                <CardContent style={styles.noResults}>
-                                    <Typography variant="body1">
-                                        {translate(
-                                            'ra.navigation.no_more_results',
-                                            { page }
-                                        )}
-                                    </Typography>
-                                </CardContent>
-                            )}
-                        {pagination &&
-                            React.cloneElement(pagination, controllerProps)}
-                    </div>
-                ) : (
-                    <CardContent className={classes.noResults}>
-                        <Typography variant="body1">
-                            {translate('ra.navigation.no_results')}
-                        </Typography>
-                    </CardContent>
-                )}
+                <div key={version}>
+                    {children &&
+                        React.cloneElement(children, {
```

Comment:
Isn't it a breaking change to not handle the no results case anymore ?

Reply:
If you use a custom datagrid, it is a small breaking change because it will show the datagrid headers with empty content. The ra datagrid has been updated to show nothing in this case. And in the case of the demo, the custom image grid for posters still works as before... because it has no headers.

If you use a custom pager, it is a small breaking change because it will not show the out of bounds message when filtering too much or going to a non-existing page. The ra Pagination has been updated to show the messages in this case. My opinion is that custom pagination components are rare, so we should no worry too much about it.

So it's not a BC break if you use the ra components. But we'll need to mention it in the blog post of the release.